### PR TITLE
Hw 5: Books app w/ store model, Devise, ActionMailer for new book & Course Tracker app

### PR DESCRIPTION
**Pull request by ingachen**

### CODE REVIEW

**Comment by Borgaard** [comment]
@ingachen looks like we have some merge conflicts. I can't merge this until the conflicts are resolved. Would you be able to meet before class? Here's what we'll do: `git commit`, then pull the `bewd11-homework` repo down to your local machine. Any merge conflicts that show up we'll fix, then commit and push your homework again.

**Comment by ingachen** [comment]
Ok I'll be there before class!

File: IngaChen_ingachen/hw_5/Books/app/controllers/application_controller.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+class ApplicationController < ActionController::Base
+  # Prevent CSRF attacks by raising an exception.
+  # For APIs, you may want to use :null_session instead.
+  protect_from_forgery with: :exception
+
+  before_action :authenticate_user!
```
Comment (Borgaard) [inline IngaChen_ingachen/hw_5/Books/app/controllers/application_controller.rb:6]: Excellent work keeping your code DRY, @ingachen! :+1:

File: IngaChen_ingachen/hw_5/Books/app/mailers/user_mailer.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+class UserMailer < ActionMailer::Base
+  default from: "notifications@booksapp.com"
+
+  def book_added(user, book)
+  	@user = user
+  	@book = book
+  	mail(to: "inga.h.chen@gmail.com", subject: "A new book has been added!")
+  end
+
```
Comment (Borgaard) [inline IngaChen_ingachen/hw_5/Books/app/mailers/user_mailer.rb:9]: :+1:

File: IngaChen_ingachen/hw_5/Books/config/routes.rb [hunk 3]
```diff
@@ -0,0 +1,62 @@
+Rails.application.routes.draw do
+  devise_for :users
+  root 'books#index'
+
+  resources :books do 
+    resources :stores
+  end
+end
```
Comment (Borgaard) [inline IngaChen_ingachen/hw_5/Books/config/routes.rb:8]: Clean and concise routes!

**Pull request merged**